### DISCUSSION TITLE
Relay via spotify (cast to HA and select internal audio as source)

### DISCUSSION THREAD

**Discussion by mortalone** [music-assistant/support#465] [Feature requests and ideas]
So. i was wondering if it would be possible to use home asistant as a relay og centerpoint for music streams and so on.

in my own usecase. i have a raspberrypi4 with a touchscreen running android. and spotify. but i am not able to utilize the awesome features of home assistant, i dan select to cast to home assistant. but i am unable to use the stream in home assistant.

i would be neat if i could stream or even better bluetooth to home assistant instance. and be able to select the internal stream from tast TO home assistant as a source. and in home assistant treat it as a source. and do all the cool things with the stream like outputting it based on movement or seamless and so on.

i hope i explained my wish clearly. if not. please ask.

**Comment by raress96** [comment]
Just created a similar discussion, but specifically for Google Cast:
https://github.com/music-assistant/hass-music-assistant/discussions/491

I think your discussion is a broader one on using Music Assistant as a relay, which will be very useful and like this idea.

**Comment by mortalone** [comment]
Bump ?

**Comment by OzGav** [comment]
Is this the same request? https://github.com/music-assistant/hass-music-assistant/discussions/625

**Reply by mortalone** [reply]
no. i want to cast to or stream to home assistant/music assistant. 
wich then in turn streams/forwards the music to a predefined speaker.

so in reality i always only have to select homeassistant and then home assistant wil figure out where to put the music and so forth.

**Reply by OzGav** [reply]
Not sure where this will sit on the roadmap but we will see how many up votes it gets!

**Reply by OzGav** [reply]
Although I don’t understand why you don’t just add Spotify as a source to MA which would negate the need  for the relay?

**Reply by mortalone** [reply]
Then i have to control spotify via HA/MA

i want to be able to control spotify from either my wallmounted screen or a phone.

There are multiple reasons why i dont want to control spotify from HA/MA.
1. when searching. the results are very limited (search for queen, artist and i get a total list of 10 tracks.)
2. no random features/discover and so forth
3. this im not total sure of. but does MA qualify for a kiosk setup ? you know with no control other than the player.

**Reply by OzGav** [reply]
OK thanks understand your use case. Will have to see what @marcelveldt says about priority for this. As for 3. you can access the MA interface directly by http://your.ip:8123/lib/mass and therefore if you had something running in Kiosk mode then that would be the only thing someone would see.

**Reply by mortalone** [reply]
that is a nice url. but it stil only returns 10 results when searching. example. searching for queen. selecting the artist. and only 10 tracks are shown. and i dont see any way to show more tracks.

**Reply by OzGav** [reply]
I think to keep the UI responsive MA does not try and load all Spotify tracks but will load anything you have in your Spotify library/favourites. However if you go to albums I think you will find them all there. At that point you can favourite albums or tracks within those albums and then those tracks will then appear.

**Comment by OzGav** [comment]
This is an old v1 discussion
